### ISSUE TITLE
[Bug] fused_recurrent_kda reads the decay tensor `g` out of bounds when the head dim is not a power of two

### ISSUE THREAD

**Issue by truong-v** [fla-org/flash-linear-attention#1134]
### Checklist

- [x] I have checked [FAQs](https://github.com/fla-org/flash-linear-attention/blob/main/FAQs.md) and existing issues for similar problems
- [x] Please report this bug in English to ensure wider understanding and support

### Describe the Bug

`fused_recurrent_kda_fwd_kernel` walks the head dim in a block of `BK = triton.next_power_of_2(K)`, and every load in its step loop is masked with `mask_k = o_k < K` — `q`, `k`, `dt_bias`, and `v`/`beta` with `mask_v` — except the load of the log-decay `g`, which is unmasked. So for any `K` that is not a power of two, each `(b, t, hv)` step reads `BK - K` floats past its row of `g`, and the last step reads past the end of the allocation itself.

The lanes beyond `K` do not change the result under exact arithmetic (the matching `b_h` rows start at zero, and `b_k` is masked to zero, so they stay zero), but what they read is fed to `exp`: if what sits after `g` is large, `exp` overflows and `0 * inf` turns the whole `b_h` block — and with it the output row — into `NaN`. `K = 60` and `K = 100` are both in this repo's own KDA test matrix, so a non-power-of-two head dim is a supported, exercised configuration; the suite still misses this because for every step except the last the out-of-bounds lanes land inside the *next* row of `g`, where log-decay values are negative and nothing propagates to the output.

https://github.com/fla-org/flash-linear-attention/blob/033b19e81239b13971a410e55dd6c178d430b9d4/fla/ops/kda/fused_recurrent.py#L151-L159

### Steps to Reproduce the Bug

```python
import torch
import torch.nn.functional as F
from fla.ops.kda import fused_recurrent_kda

B, T, H, K, V = 1, 4, 2, 60, 32          # BK = next_power_of_2(60) = 64
torch.manual_seed(42)
q = F.normalize(torch.randn(B, T, H, K, device='cuda'), p=2, dim=-1)
k = F.normalize(torch.randn(B, T, H, K, device='cuda'), p=2, dim=-1)
v = torch.randn(B, T, H, V, device='cuda')
beta = torch.randn(B, T, H, device='cuda').sigmoid()
g = F.logsigmoid(torch.randn(B, T, H, K, device='cuda', dtype=torch.float32))

# put `g` at the front of a larger buffer, so the 4 lanes read past its end are known
buf = torch.full((g.numel() + 64,), 1e30, device='cuda', dtype=torch.float32)
g_padded = buf[:g.numel()].view_as(g)
g_padded.copy_(g)
assert g_padded.is_contiguous()

o, _ = fused_recurrent_kda(q=q, k=k, v=v, g=g_padded, beta=beta)
print('non-finite outputs:', int((~torch.isfinite(o)).sum()), 'of', o.numel())
```

```
non-finite outputs: 32 of 256
```

With an ordinary `g` the reads themselves are visible under `compute-sanitizer --tool memcheck` and `PYTORCH_NO_CUDA_MEMORY_CACHING=1`: four invalid 4-byte reads at `fla/ops/kda/fused_recurrent.py:159`, one per element beyond `K = 60` (threads 60–63), landing 1/5/9/13 bytes after `g`'s 1,920-byte allocation (`ERROR SUMMARY: 8 errors`).

### Expected Behavior

No access outside `g`: the `g` load masked with `mask_k`, like every other load in the same loop — `mask_k` is already in scope a few lines above.

### Environment Information

1. Torch: 2.13.0+cu130
2. Triton: 3.7.1
3. Commit tested: 033b19e8 (current main), Python 3.12, NVIDIA B200, driver CUDA 13.2
